features/mesh/dataplanes/DataplanePolicies.feature

```diff
@@ -1,61 +1,339 @@
+        KUMA_DATAPLANERULE_COUNT: 1
+        KUMA_DATAPLANETORULE_COUNT: 3
+        KUMA_DATAPLANEFROMRULE_COUNT: 1
+        """
+      And the URL "/meshes/default/dataplanes/dataplane-1/_rules" responds with
+        """
+        body:
+          rules:
+            - type: MeshTimeout
+              fromRules:
+                - inbound:
+                    port: 8080
+                    tags:
+                      kuma.io/service: foo
+                  rules:
+                    - matchers:
+                        - key: kuma.io/service
+                          not: false
+                          value: one
+                        - key: kuma.io/service
+                          not: true
+                          value: two
+                      origin:
+                        - mesh: default
+                          name: default
+                          type: MeshTimeout
+                        - mesh: default
+                          name: override
+                          type: MeshTimeout
+                      conf:
+                        connectionTimeout: 20s
+                        idleTimeout: 20s
+                        http:
+                          requestTimeout: 5s
+              toRules:
+                - matchers:
+                    - key: kuma.io/service
+                      not: false
+                      value: foo
+                  origin:
+                    - mesh: default
+                      name: default
+                      type: MeshTimeout
+                    - mesh: default
+                      name: override
+                      type: MeshTimeout
+                  conf:
+                    connectionTimeout: 2s
+                    idleTimeout: 20s
+                    http:
+                      requestTimeout: 15s
+                - matchers:
+                    - key: kuma.io/service
+                      not: false
+                      value: bar
+                  origin:
+                    - mesh: default
+                      name: default
+                      type: MeshTimeout
+                    - mesh: default
+                      name: override
+                      type: MeshTimeout
+                  conf:
+                    connectionTimeout: 2s
+                    idleTimeout: 20s
+                    http:
+                      requestTimeout: 20s
+                - matchers:
+                    - key: kuma.io/service
+                      not: true
+                      value: bar
+                    - key: kuma.io/service
+                      not: true
+                      value: foo
+                  origin:
+                    - mesh: default
+                      name: default
+                      type: MeshTimeout
+                  conf:
+                    connectionTimeout: 2s
+                    idleTimeout: 20s
+                    http:
+                      requestTimeout: 10s
+        """
```

Comment:
Do we need to mock out all of this data?

Reply:
Most of it, yes. The only bits that we don’t test against are the `conf`s so they could technically be random.

It’s already pretty limiting to need to resort to four different env entries to control the test data here. There are things that can’t be expressed in tests like this like an empty `matchers` array in one rule while having two `matchers` items in another. This always happens as soon as an API has arrays within arrays that need controlling in tests. You’re limited to one `KUMA_*_COUNT` variable and every nested array will have to use it.